rust-escrow/dao-escrow/src/lib.rs

```diff
@@ -0,0 +1,185 @@
+}
+
+impl Default for DaoEscrow {
+    fn default() -> Self {
+        env::panic_str("DaoEscrow should be initialized before usage")
+    }
+}
+
+#[near_bindgen]
+impl DaoEscrow {
+    #[init]
+    pub fn new(dao_factory_account: AccountId) -> Self {
+        assert!(!env::state_exists(), "The contract is already initialized");
+        Self {
+            daos: UnorderedMap::new(b"r".to_vec()),
+            dao_factory_account: dao_factory_account,
+        }
+    }
+
+    pub fn daos_of(&self, country_code: String) -> u128 {
+        match self.daos.get(&country_code) {
+            Some(country) => country,
+            None => 0,
+        }
+    }
+
+    fn get_dao_config(&self, name: String, accounts: String) -> Vec<u8> {
+        format!(r#"{{ "policy": {{ "roles": [ {{ "name": "Everyone", "kind": {{ "Group": [ {} ] }}, "permissions": [ "*:Finalize", "*:AddProposal", "*:VoteApprove", "*:VoteReject", "*:VoteRemove" ], "vote_policy": {{}} }}, {{ "name": "all", "kind": "Everyone", "permissions": [ "*:AddProposal" ], "vote_policy": {{}} }} ], "default_vote_policy": {{ "weight_kind": "RoleWeight", "quorum": "0", "threshold": [ 1, 2 ] }}, "proposal_bond": "100000000000000000000000", "proposal_period": "604800000000000", "bounty_bond": "100000000000000000000000", "bounty_forgiveness_period": "604800000000000" }}, "config": {{ "name": "{}", "purpose": "", "metadata": "" }} }}"#, accounts, name).into_bytes()
+    }
+    
+    fn get_deposit_accounts(&self, deposits: Vec<(AccountId, Balance)>) -> String {
+        let mut accounts = format!(r#""{}""#, env::current_account_id().to_string());
+
+        for i in &deposits {
+            accounts += ", ";
+            accounts += &format!(r#""{}""#, &i.0.to_string().to_string());
+        }
+
+        accounts
+    }
+
+    #[payable]
+    pub fn create_dao(&mut self, country_code: String, deposits: Vec<(AccountId, Balance)>) -> Promise {
+        let mut daos_by_country = self.daos_of(country_code.clone());
+        daos_by_country = daos_by_country.wrapping_add(1);
+        
+        // Get Parameters
+        let deposit_accounts = self.get_deposit_accounts(deposits);
+        let dao_name = format!("ce_{}_{}", country_code.clone(), daos_by_country);
+        let args = self.get_dao_config(dao_name.clone(), deposit_accounts);
+
+        // Contract Call
+        ext_dao_factory::create(
+            dao_name.to_string(),
+            Base64VecU8(args),
+            self.dao_factory_account.clone(),
+            env::attached_deposit(),
+            GAS_FOR_CREATE,
+        ).then(ext_self::on_create_callback(
+            country_code,
+            daos_by_country,
+            env::current_account_id(),
+            0,
+            GAS_FOR_CREATE_CALLBACK,
+        ))
+    }
+
+    pub fn on_create_callback(&mut self, country_code: String, daos_by_country: u128) -> bool {
+        assert_eq!(
+            env::promise_results_count(),
+            1,
+            "ERR_CALLBACK_METHOD"
+        );
```

Comment:
What is this actually asserting? just curious 🤔

Reply:
It's asserting that the call was made as a callback.